hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/PoolAlignmentContext.java

```diff
@@ -64,7 +64,11 @@ public void updateResponseState(RpcHeaderProtos.RpcResponseHeaderProto.Builder h
    */
   @Override
   public void receiveResponseState(RpcHeaderProtos.RpcResponseHeaderProto header) {
-    sharedGlobalStateId.accumulate(header.getStateId());
+    if (header.getStateId() == 0 && sharedGlobalStateId.get() > 0) {
+      sharedGlobalStateId.reset();
+    } else {
+      sharedGlobalStateId.accumulate(header.getStateId());
```

Comment:
I have a naive question: What protects us here from the state where `header.getStateId() > 0 && header.getStateId() < sharedGlobalStateId`?

It seems like if this case were to occur then sharedGlobalStateId would go backwards.

Reply:
The sharedGlobalStateID is created as [`new LongAccumulator(Math::max, Long.MIN_VALUE)`](https://github.com/apache/hadoop/blob/trunk/hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterStateIdContext.java#L111)
So accumulate either keeps the current value or moves it forward.

Reply:
Ah makes sense. Thanks.